python/GafferSceneUI/AttributeQueryUI.py

```diff
@@ -36,6 +36,75 @@
 
 import Gaffer
 import GafferScene
+import GafferUI
+import imath
+import IECore
+import functools
+import uuid
+
+from GafferSceneUI._GafferSceneUI import __setupMenuTitle as setupMenuTitle
+from GafferSceneUI._GafferSceneUI import __setupMenuNames as setupMenuNames
+from GafferSceneUI._GafferSceneUI import __setupFromMenuName as setupFromMenuName
+
+class _SetupButton( GafferUI.Widget ) :
+
+	def __init__( self, node ) :
+
+		self.__node = node
+		self.__row = GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal )
+
+		GafferUI.Widget.__init__( self, self.__row )
+
+		with self.__row :
+
+				GafferUI.Spacer( imath.V2i( GafferUI.PlugWidget.labelWidth(), 1 ) )
+
+				menuButton = GafferUI.MenuButton(
+					image = "plus.png",
+					hasFrame = False,
+					menu = GafferUI.Menu(
+						Gaffer.WeakMethod( self.__menuDefinition ),
+						title = setupMenuTitle()
+					)
+				)
+
+				GafferUI.Spacer( imath.V2i( 1 ), imath.V2i( 999999, 1 ), parenting = { "expand" : True } )
+
+		node.childAddedSignal().connect( Gaffer.WeakMethod( self.__updateVisibility ), scoped = False )
+		node.childRemovedSignal().connect( Gaffer.WeakMethod( self.__updateVisibility ), scoped = False )
+
+		self.__updateVisibility()
+
+	def __menuDefinition( self ) :
+
+		menuDefinition = IECore.MenuDefinition()
+
+		for name in setupMenuNames() :
+			if not name.split('/')[~0] :
+				menuDefinition.append(
+					"/" + name + uuid.uuid4().hex,
+					{
+						"divider" : True
+					}
+				)
+			else :
+				menuDefinition.append(
+					"/" + name,
+					{
+						"command" : functools.partial( Gaffer.WeakMethod( self.__setup ), name )
+					}
+				)
+
+		return menuDefinition
+
+	def __setup( self, name ) :
+
+		setupFromMenuName( self.__node, name )
```

Comment:
This would need to be wrapped in an UndoScope as well (see comment in AttributeQueryUI.cpp), but that won't matter if we let AttributeQueryUI.cpp create the menu for us (because we'll have one there).